### ISSUE TITLE
List of "known issues" to help direct contributors

### ISSUE THREAD

**Issue by mwarrenus** [JasnaPaka/mozbackup#1]
The [MozBackup support page](http://mozbackup.jasnapaka.com/support.php) says that

> … There are known issues and there is no time on my side to fix all issues and develop new features

Could you please create issues here in Github to help direct community contributions and/or provide technical details about the issues.  Thank you for creating MozBackup and publishing its source code!

**Comment by TeoTwawki** [comment]
Yeah can't contribute fixes, or fork and maintain it ourselves, if these known issues aren't known to anyone but the original dev.

**Comment by MikkCZ** [comment]
AFAIK as MozBackup has not been developed for a while already, it will have problems with backing up and restoring all the files on all system versions. @JasnaPaka might tell you more details, but I think the biggest know issues is it's a bit old.

From my point of view, in days of Firefox having Sync feature, and SUMO pretty neat documentation of how profiles works and can be migrated, there is no urgent need for MozBackup anymore. See https://support.mozilla.org/en-US/kb/back-and-restore-information-firefox-profiles, https://support.mozilla.org/en-US/kb/recovering-important-data-from-an-old-profile or https://support.mozilla.org/en-US/kb/moving-thunderbird-data-to-a-new-computer. That's what MozBackup does.

**Comment by TeoTwawki** [comment]
Don't care, still want to know what these "known issues" actually are.

\<Insert some argument about change is bad / we don't like change / wnat to keep using old things here.\> 😃

**Comment by scristaldi** [comment]
Has this gone any further? Any problems/issues list?
Source documentation rather poor and no documentation I can find.

**Comment by InterruptDriven** [comment]
Just now discovered that MozBackup has been abandoned. That's too bad. 

As a Delphi "coder" I'd be interested in solving some of the problems in order to revive MozBackup.

**Comment by shubnigg** [comment]
I have this problems when I try to backup Thunderbird  v52.2.1 (32-bit) with mozbackup version 1.5.2 beta (same with older version)

> access violation at address 004050E6 in module mozbackup.exe. read of address 00000012.

**Comment by Stephen-Leibowitz** [comment]
I am not experienced in Pascal, but here is a short list of suggestions.

1) The comments and identifiers are a mixture of English and Czech. A full
conversion to a single language (probably English) would be ideal. A more
modest approach would be to focus on converting the comments and local
identifiers in those program sections that are being revised for other
reasons. Google Translate could help.

2) Pascal Analyzer could be useful. There is the full version and a Lite
(freeware) version:
http://www.peganza.com/download.html
http://www.peganza.com/introducing-pascal-analyzer-lite.html

3) Perhaps update DelphiZip from version 1.90 to 1.92.

4) Perhaps update to a newer version of SQLite.

**Comment by Stephen-Leibowitz** [comment]
I am not experienced in Pascal, but here is a short list of suggestions.

1) The comments and identifiers are a mixture of English and Czech. A full conversion to a single language (probably English) would be ideal. A more modest approach would be to focus on converting the comments and local identifiers in those program sections that are being revised for other reasons. Google Translate could help.

2) Pascal Analyzer could be useful. There is the full version and a Lite (freeware) version:
http://www.peganza.com/download.html
http://www.peganza.com/introducing-pascal-analyzer-lite.html

3) Perhaps update DelphiZip from version 1.90 to 1.92.

4) Perhaps update to a newer version of SQLite.

**Comment by FreeDOS-LA** [comment]
> 
> 
> I am not experienced in Pascal, but here is a short list of suggestions.
> 
>     1. The comments and identifiers are a mixture of English and Czech. A full conversion to a single language (probably English) would be ideal. A more modest approach would be to focus on converting the comments and local identifiers in those program sections that are being revised for other reasons. Google Translate could help.

Yeah, all the Czech comments and identifier names make the source code close to useless...
> 
>     2. Pascal Analyzer could be useful. There is the full version and a Lite (freeware) version:
>        http://www.peganza.com/download.html
>        http://www.peganza.com/introducing-pascal-analyzer-lite.html
> 
The freeware version is totally useless. It randomly removes stuff from the output/analysis, which makes it hard to judge if the program actually works...

**Comment by JasnaPaka** [comment]
Greetings to everyone after many years of my inactivity on this project. First of all, I apologize for never answering your questions. At some point in my life, I left further development of MozBackup to pursue other activities. Even the published code was wrong - it could not be compiled, contained Czech comments, etc. In retrospect, I perceive that even the quality of the source code is nowhere near what I expect today. I apologize.

And now the better news. I would like to resume the development of MozBackup. It seems to me that despite backup services like Firefox Sync or integrated backup/restore in Thunderbird, this product makes sense.

The plan now is as follows:
* I now call MozBackup 1.x "legacy". However, I am now gradually preparing version 1.6, where I would like to address support for everything new that has happened in the last X years. Don't expect much new stuff here, rather I'd like to recreate a working product that you can use.
* Alongside this I will be working on "MozBackup 2.0", which should be new software. It will expand the scope a bit and will be available for other platforms like macOS/Linux. For future information, keep an eye on the MozBackup homepage, where there is an option to subscribe to changes via RSS.

I will now close this entry. Please report any bugs or suggestions as new separate bugs or use the discussion here on GitHub for general discussion.
